### PR TITLE
Changes up carbon EMP handling

### PR DESCRIPTION
**Pull request by Maia-J**
## About The Pull Request

The organ refactor 6 months ago changed up organs, making them be actually inside bodyparts and the mob. This introduced the bug of all emp effects being called twice on everything thanks to how /mob/living handles it (Fixed in this PR), but also some new stuff:

Made bodyparts handle organ EMPs. This also means we can now have support for bodyparts with EMP_PROTECT_CONTENTS protecting the organs inside.
Made a new proc for bodyparts once they are successfully hit by an emp. Makes it much easier to add overrides for the behaviour.
Fixed emps hitting bodyparts twice

## Why It's Good For The Game

It lays a groundwork for some interesting limb concepts, and makes organ emps make a bit more sense.

Fixes a bug that laid unnoticed for 6 months.

## Changelog

:cl:
fix: EMPs on carbons no longer happen twice
code: Moves organ emps under bodyparts, changes how bodyparts handle emp effects
/:cl:

### CODE REVIEW

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 1]
```diff
@@ -1313,9 +1313,21 @@
 // Note: Does NOT return EMP protection value from parent call or pass it on to subtypes
```
Comment (MrMelbert) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1313]: Update or remove the comment

File: code/modules/surgery/bodyparts/robot_bodyparts.dm [hunk 2]
```diff
@@ -115,18 +115,19 @@
 	damage_examines = list(BRUTE = ROBOTIC_BRUTE_EXAMINE_TEXT, BURN = ROBOTIC_BURN_EXAMINE_TEXT)
 	bodypart_flags = BODYPART_UNHUSKABLE
 
-/obj/item/bodypart/leg/left/robot/emp_act(severity)
+/obj/item/bodypart/leg/left/robot/emp_effect(severity, protection)
 	. = ..()
-	if(!. || isnull(owner))
-		return
+	if(isnull(owner))
```
Comment (MrMelbert) [inline code/modules/surgery/bodyparts/robot_bodyparts.dm:120]: Why get rid of `!.`? Same goes for all of these.

Comment (Majkl-J) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:120]: The robot limb will always be robotic, we really don't need to spend time checking the return value of the parent

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 3]
```diff
@@ -1310,12 +1310,24 @@
...
+		for(var/atom/content as anything in contents)
+			content.emp_act(severity)
+
+	if((protection & (EMP_PROTECT_WIRES | EMP_PROTECT_SELF)))
+		return protection
 
+	emp_effect(severity, protection)
+	return protection
+
+// The actual effect of EMPs on the limb. Allows children to override it however they want
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1327]:
```suggestion
/// The actual effect of EMPs on the limb. Allows children to override it however they want
```

File: code/modules/surgery/bodyparts/robot_bodyparts.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:120]: You should still keep it, there's no reason not to and it saves future time if we add some other `return FALSE` condition to `emp_effect`

**Pull request merged**